src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -548,28 +550,18 @@ protected void shardOperationOnReplica(ShardId shardId, BulkShardRequest request
 
                     if (indexRequest.opType() == IndexRequest.OpType.INDEX) {
                         Engine.Index index = indexShard.prepareIndex(sourceToParse, indexRequest.version(), indexRequest.versionType(), Engine.Operation.Origin.REPLICA, request.canHaveDuplicates() || indexRequest.canHaveDuplicates());
-                        if (index.parsedDoc().dynamicMappingsUpdate() != null) {
-                            if (indexService.index().name().equals(RiverIndexName.Conf.indexName(settings))) {
-                                // mappings updates on the _river are not validated synchronously so we can't
-                                // assume they are here when indexing on a replica
-                                indexService.mapperService().merge(indexRequest.type(), new CompressedString(index.parsedDoc().dynamicMappingsUpdate().toBytes()), true);
-                            } else {
-                                throw new ElasticsearchIllegalStateException("Index operations on replicas should not trigger dynamic mappings updates: [" + index.parsedDoc().dynamicMappingsUpdate() + "]");
-                            }
+                        Mapping update = index.parsedDoc().dynamicMappingsUpdate();
```

Comment:
should we have a utility method in the superclass for this? or in a Utils class? we have this code copied quiet often?